lib/services/android-plugin-build-service.ts

```diff
@@ -0,0 +1,385 @@
+import * as path from "path";
+import * as shell from "shelljs";
+
+const xml2js = require("xml2js");
+
+export class AndroidPluginBuildService implements IAndroidPluginBuildService {
+
+	constructor(private $fs: IFileSystem) { }
+
+	private static ANDROID_MANIFEST_XML = "AndroidManifest.xml";
+	private static INCLUDE_GRADLE = "include.gradle";
+	private static MANIFEST_ROOT = {
+		$: {
+			"xmlns:android": "http://schemas.android.com/apk/res/android"
+		}
+	};
+	private static ANDROID_PLUGIN_GRADLE_TEMPLATE = "gradle-plugin";
+	// private static DEFAULT_PLUGIN_NAME = "myplugin";
+
+	private getAndroidSourceDirectories(source: string): Array<string> {
+
+		const getLsStats = this.$fs.getLsStats;
+		///// Helpe methods to operate with android sourceset files (res, assets, java, etc.)
+		function isAndroidSourceDirectory(src: string) {
+			const sourceNameParts = src.split(path.sep);
+			const dirName = sourceNameParts[sourceNameParts.length - 1];
+			return (
+				getLsStats(src).isDirectory() &&
+				(dirName === "res" ||
+					dirName === "java" ||
+					dirName === "assets" ||
+					dirName === "jniLibs")
+			);
+		}
+
+		return this.getDirectories(source).filter(isAndroidSourceDirectory);
+	}
+
+	private getManifest(platformsDir: string) {
```

Comment:
Can you use `platformsData.getPlatformData().configurationFilePath`?

Reply:
this is the plugin's platforms dir, not the project